DirectProgramming/C++/CompilerInfrastructure/OpenMP_Offload_Features/src/usm_and_composability_with_dpcpp.cpp

```diff
@@ -0,0 +1,66 @@
+//==============================================================
+// Copyright © 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <omp.h>
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <iostream>
+
+using namespace sycl;
+
+extern "C" void *omp_target_get_context(int);
+
+#ifdef OCL_BACKEND
+#pragma omp requires unified_shared_memory
+#endif
+
+int main() {
+  const unsigned kSize = 200;
+  int d = omp_get_default_device();
+#ifdef OCL_BACKEND
+  sycl::queue q(
+      sycl::context(static_cast<cl_context>(omp_target_get_context(d))),
+      sycl::gpu_selector());
+#else
+  sycl::queue q;
+#endif
+  std::cout << "SYCL: Running on "
+            << q.get_device().get_info<sycl::info::device::name>() << "\n";
+  if (!q.get_device()
+           .get_info<sycl::info::device::usm_shared_allocations>()) {
+    std::cout << "SYCL: USM is not available\n";
+    return 0;
+  }
+  auto Validate = [](int *data) {

```

Comment:
You have int main(), but you are returning a char* value from Validate and then return this from main.

This should be type correct.   You should be printing passed/failued, and then returning 0/1.

Reply:
Hi @pmpeter1 ,
In the current code,  TestOmp and TestDPCPP are returning the char* value from Validate which gets printed in main(). main() only returns 0.

Reply:
@pmpeter1 are you we good with this reply?